### ISSUE TITLE
Excluded items aren't replaced for top-k recommendations

### ISSUE THREAD

**Issue by dorukkilitcioglu** [fidelity/mab2rec#5]
When the `item_eligibility` parameter is used to control the recommendations for customers while a `top-k` is specified, the `score` pipeline returns less than `top-k` items if one of the items that would've been recommended to the user is excluded. 

Example:
```python
k = 4
items = [1, 2, 3, 4, 5]
scores = [0.9, 0.8, 0.7, 0.6, 0.5]
exclusion_list = [1]
recs = [2, 3, 4, 5] or [2, 3, 4]?
```

M2R returns `[2, 3, 4]`, whereas it can backfill from non-excluded items and return `[2, 3, 4, 5]`.

Originally reported by @nateewall, you can add more to the issue if there's any other issues you've seen with this behavior.

**Comment by skadio** [comment]
That's a good catch! Thank you for sharing! Let's make sure to attribute the fix to @nateewall in the library docs

**Comment by dorukkilitcioglu** [comment]
This has been resolved as of #6. Thanks @nateewall for reporting the issue and @bkleyn for the fast return on the PR!
